src/api/chat.ts

```diff
@@ -0,0 +1,74 @@
+import { getDatabase, ref, update, push, child, get, serverTimestamp } from "firebase/database";
+import log from "loglevel";
+import * as util from './util'
+import { Message, Chat, MessageParams } from '../types';
+
+export interface DbChat { 
+  id: string,
+  messages?: Record<string, Message>
+}
+
+export default {
+  async get(groupId: string): Promise<Chat | null> {
+    const db = getDatabase()
+    const chatRef = ref(db, `groups/${groupId}/chat`);
+    const snapshot = await get(chatRef);
+    const val = snapshot.val() as DbChat;
+    if (!val) null
+    const chat: Chat = { groupId: val.id }
+    // WHEN THERE IS NO MESSAGE, RETURN GROUPID ONLY
+    if (!val.messages) return chat
+    chat.messages = util.fbMapToIdArr(val.messages)
+
+    return chat
+  },
+
+  async create(groupId: string): Promise<Chat> {
+    const db = getDatabase()
+    const updates = {} as Record<string, any>
+    const chatData: DbChat = {
+      id: groupId,
```

Comment:
Here is what I kind of doesn't really like. Should this just be groupId for now?

Reply:
updated `groupId` to `chatId`